src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -2,37 +2,81 @@ namespace NServiceBus
 {
     using System;
     using System.Diagnostics;
-#if NETSTANDARD
-    using System.Runtime.InteropServices;
-#endif
     using System.Text;
     using System.Threading;
     using System.Threading.Tasks;
     using Logging;
     using Particular.Licensing;
+#if NETSTANDARD
+    using System.Runtime.InteropServices;
+
+#endif
 
     class LicenseManager
     {
         internal bool HasLicenseExpired => result?.HasExpired ?? true;
 
+        public LicenseManager(Func<DateTime> utcDateTimeProvider)
```

Comment:
Why is this a `Func<DateTime>` rather than a `DateTime`?

Reply:
hm, good point. Strictly speaking this wouldn't be necessary, assuming the license manager gets initialized soon after being created. But since that would require knowledge of the exact usage pattern of the LicenseManager I'd vote to keep it as a Func to allow more recent date at actual execution time.

Reply:
Why is it important that the time is not before the execution of the license check?

Reply:
Scratch that last question. I get it. If the license check is way after the initialisation (for whatever bizarre reason), we want it to be as late as possible, i.e. license may expire in between.